zephyr/esp32s3/src/wifi/esp_wifi_adapter.c

```diff
@@ -36,7 +37,11 @@ LOG_MODULE_REGISTER(esp32_wifi_adapter, CONFIG_WIFI_LOG_LEVEL);
 #include "soc/dport_reg.h"
 #include "driver/adc2_wifi_private.h"
 
-K_THREAD_STACK_DEFINE(wifi_stack, 8192);
+#define WIFI_PKT_SIZE	1700
+K_HEAP_DEFINE(wifi_heap, WIFI_PKT_SIZE*MAX(CONFIG_NET_BUF_RX_COUNT, CONFIG_NET_BUF_TX_COUNT));
```

Comment:
The only issue I see is that it is duplicating RAM usage as this heap size is already allocated in net-stack (`net_pkt.c`). So adding this will increase RAM usage by a few 10~15kb (ok, depends on the configs).  Anyway, I understand this fixes the issue you have been facing. I will try to reproduce again to evaluate it.

Reply:
Yes, the network packets and extra storage usage is not great, but it has always been the case -- it was just hidden in the system heap.  If you look at `esp32_wifi_send()`, it copies the TX packet into `data->frame_buf` to create a flat buffer and then that is passed to `esp_wifi_internal_tx()` which allocates a packet using `wifi_malloc()` and copies the data again.  Not ideal, but I didn't see any way to improve it. 

An alternative implementation to the `K_HEAP_DEFINE()` would be to allocate the `wifi_heap` from the system heap.  The issue I ran into is that if the SPIRAM heap is enabled, then this allocation is from SPIRAM and WiFi fails to connect.